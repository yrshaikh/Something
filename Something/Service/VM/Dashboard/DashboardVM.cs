using Service.VM.Common;
using Service.VM.Enums;

namespace Service.VM.Dashboard
{
    public class DashboardVM : BasePageVM
    {
        public override int PageTypeId { get; set; } = (int)PageTypeEnum.Home;
    }

}
