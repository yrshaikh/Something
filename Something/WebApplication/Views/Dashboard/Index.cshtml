@using WebApplication.Utils;
@using Service.VM.Dashboard;
@{
    ViewData["Title"] = "Index";
}

<div id="root"></div>

<script type="text/javascript">
    window.params = @Html.Raw(JsonSerializer.Serialize<DashboardVM>(Model));
</script>

<script src="~/dist/dashboard.bundle.js"></script>