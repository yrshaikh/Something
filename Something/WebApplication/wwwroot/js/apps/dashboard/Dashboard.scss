@import "../Global.scss";
@import "../Variables.scss";

.Dashboard {
    
}