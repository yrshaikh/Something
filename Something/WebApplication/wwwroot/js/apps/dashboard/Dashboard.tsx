import * as React from "react";

import "./Dashboard.scss";
import { IDashboardProps } from "./types/IDashboardProps";

export const Dashboard : React.SFC<IDashboardProps> = () => {
    return <div className="Dashboard">
        Nothing here. Yet.
    </div>;
}
