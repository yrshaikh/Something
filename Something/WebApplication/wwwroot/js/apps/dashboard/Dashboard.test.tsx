import "jest";
import * as React from "react";
import { shallow } from "enzyme";
import enzymeToJson from "enzyme-to-json";

import { Dashboard } from "./Dashboard";

describe("<Dashboard />", () => {
    it("should render correctly", () => {
        const wrapper = shallow(<Dashboard />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it("has expected contents on it", () => {
        const wrapper = shallow(<Dashboard />);
        expect(wrapper.find('.Dashboard').length).toBe(1);
        expect(wrapper.find('.Dashboard').text()).toBe('Nothing here. Yet.');
    });
});
