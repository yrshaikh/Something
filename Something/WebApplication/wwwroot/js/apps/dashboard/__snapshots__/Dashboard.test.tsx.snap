// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should render correctly 1`] = `
<div
  className="Dashboard"
>
  Nothing here. Yet.
</div>
`;
